packages/shared/src/components/post/PostActions.tsx

```diff
@@ -77,6 +81,31 @@ export function PostActions({
     await toggleDownvote({ payload: post, origin });
   };
 
+  const [getPost, updatePost] = useUpdateQuery(
+    postByIdQueryOptions({
+      id: post.id,
+    }),
+  );
+
+  useQuerySubscription(({ variables }) => {
+    const { type, entityId } = variables;
```

Comment:
<img width="524" alt="image" src="https://github.com/user-attachments/assets/559af740-a9e7-4fb4-89b9-b112c8c72bf6" />

In contrast to `useMutationSubscription` that you have to manually type and provide `matcher` to catch mutation events this hook is fully typed automatically due to it using same creator function as the `useMutation`.